app/scripts/background.js

```diff
@@ -176,6 +178,103 @@ const sendReadyMessageToTabs = async () => {
   }
 };
 
+/**
+ * Detects known phishing pages as soon as the browser begins to load the
+ * page. If the page is a known phishing page, the user is redirected to the
+ * phishing warning page.
+ *
+ * This detection works even if the phishing page is now a redirect to a new
+ * domain that our phishing detection system is not aware of.
+ *
+ * @param {MetamaskController} theController
+ */
+function maybeDetectPhishing(theController) {
+  async function redirectTab(tabId, url) {
+    try {
+      return await browser.tabs.update(tabId, {
+        url,
+      });
+    } catch (error) {
+      return sentry?.captureException(error);
+    }
+  }
+  // we can use the blocking API in MV2, but not in MV3
+  const blocking = !isManifestV3;
+  browser.webRequest.onBeforeRequest.addListener(
+    (details) => {
+      if (details.tabId === browser.tabs.TAB_ID_NONE) {
+        return {};
+      }
+
+      const onboardState = theController.onboardingController.store.getState();
+      if (!onboardState.completedOnboarding) {
+        return {};
+      }
+
+      const prefState = theController.preferencesController.store.getState();
+      if (!prefState.usePhishDetect) {
+        return {};
+      }
+
+      // ignore requests that come from our phishing warning page
+      if (
+        details.initiator &&
+        // compare normalized URLs
+        new URL(details.initiator).toString() === phishingPageHref
```

Comment:
Oh interesting. I'll test it.

IIRC, this check was necessary because the "Continue to site" link on our phishing warning page begins the redirect before our `content_script` can tell the `phishingController` that the domain is safe.

Reply:
I've tested this and while the `sandbox` property on the iframe did cause everything to stop working, it wasn't for the reason you'd expect.

So `sandbox` actually prevents our `Continue to site` links (and other links that want to open in `_blank` like "Open this warning in a new tab") from working at all.

This is an existing issue that I think is out of scope for this PR. I _think_ the solution is for the iframe to send a message to background.js asking it to open the new tab for us (since the `sandbox` attribute without an `allow-popups` prevents it).

I encourage reviewers to attempt their own manual testing with `<iframe sandbox>`.